source/MaterialXTest/MaterialXCore/Unit.cpp

```diff
@@ -159,33 +161,53 @@ TEST_CASE("UnitDocument", "[unit]")
                                 float originalval = value->asA<float>();
                                 float convertedValue = uconverter->convert(originalval, input->getUnit(), DISTANCE_DEFAULT);
                                 float reconvert = uconverter->convert(convertedValue, DISTANCE_DEFAULT, input->getUnit());
+                                results[input->getNamePath()] = mx::Value::createValue<float>(convertedValue)->getValueString();
                                 REQUIRE((originalval - reconvert) < EPSILON);
                             }
                             else if (type == "vector2")
                             {
                                 mx::Vector2 originalval = value->asA<mx::Vector2>();
                                 mx::Vector2 convertedValue = uconverter->convert(originalval, input->getUnit(), DISTANCE_DEFAULT);
                                 mx::Vector2 reconvert = uconverter->convert(convertedValue, DISTANCE_DEFAULT, input->getUnit());
+                                results[input->getNamePath()] = mx::Value::createValue<mx::Vector2>(convertedValue)->getValueString();
                                 REQUIRE(originalval == reconvert);
                             }
                             else if (type == "vector3")
                             {
                                 mx::Vector3 originalval = value->asA<mx::Vector3>();
                                 mx::Vector3 convertedValue = uconverter->convert(originalval, input->getUnit(), DISTANCE_DEFAULT);
                                 mx::Vector3 reconvert = uconverter->convert(convertedValue, DISTANCE_DEFAULT, input->getUnit());
+                                results[input->getNamePath()] = mx::Value::createValue<mx::Vector3>(convertedValue)->getValueString();
                                 REQUIRE(originalval == reconvert);
                             }
                             else if (type == "vector4")
                             {
                                 mx::Vector4 originalval = value->asA<mx::Vector4>();
                                 mx::Vector4 convertedValue = uconverter->convert(originalval, input->getUnit(), DISTANCE_DEFAULT);
                                 mx::Vector4 reconvert = uconverter->convert(convertedValue, DISTANCE_DEFAULT, input->getUnit());
+                                results[input->getNamePath()] = mx::Value::createValue<mx::Vector4>(convertedValue)->getValueString();
                                 REQUIRE(originalval == reconvert);
                             }
                         }
                     }
                 }
             }
         }
+
+        if (!results.empty())
```

Comment:
As I don't think this method should be added, this test does not need to change naturally...